library/public/types.h

```diff
@@ -58,6 +71,54 @@ class F3D_EXPORT ratio_t
   double Value;
 };
 
+/**
+ * Describe a RGB color.
+ */
+class F3D_EXPORT color_t
+{
+public:
+  color_t() = default;
+  color_t(const std::vector<double>& vec)
+  {
+    if (vec.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided vector does not have the right size");
+    }
+    std::copy_n(vec.begin(), 3, this->RGB.begin());
+  }
+  color_t(double red, double green, double blue)
+  {
+    this->RGB[0] = red;
+    this->RGB[1] = green;
+    this->RGB[2] = blue;
+  }
+  template<std::size_t N>
+  color_t(const double (&list)[N])
+  {
+    static_assert(N == 3, "color_t needs exactly 3 elements to initialize");
+    std::copy_n(list, 3, this->RGB.begin());
+  }
+  operator std::vector<double>() const
+  {
+    return std::vector<double>(this->RGB.begin(), this->RGB.end());
+  }
+  bool operator==(const color_t& other) const
+  {
+    return this->RGB == other.RGB;
+  }
+  bool operator!=(const color_t& other) const
+  {
+    return this->RGB != other.RGB;
+  }
+  const double* data() const
```

Comment:
There's no way to change the color after construction?

Reply:
good point. Should I add a `operator[]` ?

Reply:
yes, const and not const.  
`double r()` / `double g()` / `double b()` could be useful too